gemini/evaluation/evaluating_prompts_at_scale_with_gemini_batch_prediction_api.ipynb

```diff
@@ -0,0 +1,1011 @@
+    "outputId": "019a3d3a-ddeb-425e-f9b3-62efd07938fc"
+   },
+   "outputs": [
+    {
+     "name": "stdout",
+     "output_type": "stream",
+     "text": [
+      "\u001b[2K   \u001b[90m━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━━\u001b[0m \u001b[32m5.1/5.1 MB\u001b[0m \u001b[31m17.0 MB/s\u001b[0m eta \u001b[36m0:00:00\u001b[0m\n",
+      "\u001b[?25h"
+     ]
+    }
+   ],
+   "source": [
+    "! pip install google-cloud-aiplatform --upgrade -q"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "f9nGCMsxJZ_v"
+   },
+   "source": [
+    "### Restart Colab"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 2,
+   "metadata": {
+    "colab": {
+     "base_uri": "https://localhost:8080/"
+    },
+    "id": "LIWOo23oVGmV",
+    "outputId": "c4f6cf20-894d-45ef-9837-7c4c0e0f864a"
+   },
+   "outputs": [
+    {
+     "data": {
+      "text/plain": [
+       "{'status': 'ok', 'restart': True}"
+      ]
+     },
+     "execution_count": 2,
+     "metadata": {},
+     "output_type": "execute_result"
+    }
+   ],
+   "source": [
+    "# You will see a notification of Colab crashing. It is the expected behavior.\n",
+    "import IPython\n",
+    "\n",
+    "app = IPython.Application.instance()\n",
+    "app.kernel.do_shutdown(True)"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "o_Z7Y63KJelC"
+   },
+   "source": [
+    "### Authenticate your notebook environment (Colab only)"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 1,
+   "metadata": {
+    "id": "UV7jgeu9VImB"
+   },
+   "outputs": [],
+   "source": [
+    "import sys\n",
+    "\n",
+    "if \"google.colab\" in sys.modules:\n",
+    "    from google.colab import auth\n",
+    "\n",
+    "    auth.authenticate_user()"
+   ]
+  },
+  {
+   "cell_type": "markdown",
+   "metadata": {
+    "id": "m-FTvlzlJ8lY"
+   },
+   "source": [
+    "### Define constants"
+   ]
+  },
+  {
+   "cell_type": "code",
+   "execution_count": 80,
+   "metadata": {
+    "id": "HDVVYzTISvGv"
+   },
+   "outputs": [],
+   "source": [
+    "PROJECT_ID = \"your-project-id\"\n",
+    "LOCATION = \"us-central1\"\n",
+    "\n",
+    "# Generative model.\n",
+    "MODEL_ID = \"gemini-1.5-flash-001\"\n",
+    "\n",
+    "# BigQuery tables.\n",
+    "BQ_DATASET_ID = \"text_extraction_3\"\n",
+    "BQ_DATASET = f\"{PROJECT_ID}.{BQ_DATASET_ID}\"\n",
+    "FILES_TABLE = f\"{BQ_DATASET_ID}.sports_files\"\n",
+    "PROMPTS_TABLE = f\"{BQ_DATASET}.temp_prompts\"\n",
+    "TEXT_GENERATION_TABLE_PREFIX = f\"{BQ_DATASET}.results\"\n",
+    "\n",
+    "# BigQuery views.\n",
+    "RESULTS_VIEW = f\"{BQ_DATASET}.extraction_results\"\n",
+    "EVALUATION_VIEW = f\"{BQ_DATASET}.evaluation\"\n",
+    "\n",
+    "# GCS Bucket.\n",
+    "BUCKET_NAME = \"your-bucket-name\"\n",
+    "SPORTS_FILE = f\"gs://{BUCKET_NAME}/sports_files.csv\"\n",
+    "\n",
+    "# Stop states from .\n",
+    "STOP_STATES = (\"JOB_STATE_SUCCEEDED\", \"JOB_STATE_FAILED\")"
```

Comment:
What is this needed for?

Reply:
These are stop states for checking the status of the request to Gemini Batch API. I moved them to the function that checks the status of the request.